contract_partner/models/contract.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    service_partner_id = fields.Many2one(
+        'res.partner',
+        string="Service Address")
```

Comment:
I suggest `Service Location`. It's more generic: in the same address, I could have several locations, such as "Second Floor" or "Office 4B".